velox/exec/Operator.cpp

```diff
@@ -294,6 +294,25 @@ OperatorStats Operator::stats(bool clear) {
   return ret;
 }
 
+uint32_t Operator::outputBatchRows() const {
+  const auto& queryConfig = operatorCtx_->task()->queryCtx()->queryConfig();
+
+  auto rowSize = estimatedOutputRowSize();
+  if (rowSize == kUnknownRowSize) {
+    return queryConfig.defaultOutputBatchRows();
+  }
+
+  if (rowSize * 10'000 < queryConfig.preferredOutputBatchBytes()) {
+    return 10'000; // No more than 10'000 rows
```

Comment:
Should we make this a config as well?

Reply:
@mbasmanova , I got this logic of capping # of rows by 10'000 from the TableScan::setBatchSize(). Orri once mentioned that this logic is in TableScan because # of row that TableScan can read once is no more than # of rows of a row group, 10'000 rows. Then in general case here shall we remove the cap?